runtime/autoload/help.vim

```diff
@@ -0,0 +1,40 @@
+if exists("g:loaded_help")
+  finish
+endif
+let g:loaded_help = 1
+
+function! help#topic() abort
```

Comment:
Honestly, I barely know myself since this was blatantly copied from https://github.com/tpope/vim-scriptease/blob/59a73a2415ea1b006ae3e91163ddde24e8540844/autoload/scriptease.vim#L693-L723. Initially this pull request only contained a simple `keywordprg=:help`, but then it was suggested to use vim-scriptease's more intelligent lookup.

But in short, this function looks at the word under the cursor (e.g. what syntax group it has), and on the word(s) coming before and after.